sync-diff-inspector: update required privilege for sync-diff-inspector (#21160)
pingcap/docs: sync-diff-inspector/sync-diff-inspector-overview.md

## Database privileges for sync-diff-inspector

To access table schemas and query data, sync-diff-inspector requires specific database privileges. Grant the following privileges on both the upstream and downstream databases:

> - `SELECT`: required for data comparison.

Reviewer: ```suggestion
- `SELECT`: required to compare data.
```

## Database privileges for sync-diff-inspector

To access table schemas and query data, sync-diff-inspector requires specific database privileges. Grant the following privileges on both the upstream and downstream databases:

- `SELECT`: required for data comparison.
- `RELOAD`: required to view table schemas.

>     - `RELOAD` (views table schema)

Reviewer 2: ```suggestion
```

> sync-diff-inspector needs to obtain the information of table schema and to query data. The required database privileges are as follows:

Reviewer 2: ```suggestion
To access table schemas and query data, sync-diff-inspector requires specific database privileges. Grant the following privileges on both the upstream and downstream databases:
```

> * Upstream/Downstream database

Reviewer 2: ```suggestion
```

>     - `SELECT` (checks data for comparison)

Reviewer 2: ```suggestion
- `SELECT`: required for data comparison.
```

>     - `RELOAD` (views table schema)

Author: I think we don't need `RELOAD` too?

Reviewer 3: RELOAD would be needed for FTWRL, but sync-diff did not use it currently. 

I'm not sure what "views table schema" means. `SHOW CREATE VIEW` and `` `INFORMATION_SCHEMA`.`VIEWS` `` requires "SHOW VIEW" privilege which is irrelevant to RELOAD, and sync-diff ignores views.

> **Note**: Please avoid granting the [`SHOW DATABASES`](https://dev.mysql.com/doc/refman/8.4/en/privileges-provided.html#priv_show-databases) privilege for all data sources. For MySQL data sources, ensure the [`skip_show_database`](https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_skip_show_database) system variable is set to `false` too. Otherwise, the check may fail.

Reviewer 3: ```suggestion
**Note**: Please avoid granting the [`SHOW DATABASES`](https://dev.mysql.com/doc/refman/8.4/en/privileges-provided.html#priv_show-databases) privilege on all databases (`*.*`), otherwise sync-diff-inspector will try to visit the inaccessible databases. For MySQL data sources, ensure the [`skip_show_database`](https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_skip_show_database) system variable is set to `OFF` too. Otherwise, the check may fail.
```

> **Note**: Please avoid granting the [`SHOW DATABASES`](https://dev.mysql.com/doc/refman/8.4/en/privileges-provided.html#priv_show-databases) privilege on all databases (`*.*`), otherwise sync-diff-inspector will try to visit the inaccessible databases. For MySQL data sources, ensure the [`skip_show_database`](https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_skip_show_database) system variable is set to `OFF` too. Otherwise, the check may fail.

Reviewer 2: ```suggestion
> **Note**:
> 
> - **DO NOT** grant the [`SHOW DATABASES`](/sql-statements/sql-statement-show-databases.md) privilege on all databases (`*.*`). Otherwise, sync-diff-inspector will attempt to access inaccessible databases, which causes errors.
> - For MySQL data sources, ensure that the [`skip_show_database`](https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_skip_show_database) system variable is set to `OFF`. If this variable is set to `ON`, the check might fail.
```
